lightning/src/util/ser.rs

```diff
@@ -381,6 +383,38 @@ impl Readable for BigSize {
 	}
 }
 
+/// The lightning protocol uses u16s for lengths in most cases. As our serialization framework
+/// primarily targets that, we must as well. However, because we may serialize objects that have
+/// more than 65K entries, we need to be able to store larger values. Thus, we define a variable
+/// length integer here which is backwards-compatible but treats 0xffff as "read eight more bytes".
+///
+/// To ensure we only have one valid encoding per value, we add 0xffff to values written as eight
+/// bytes, representing 0xffff as zero-eight-bytes, which 0xfffe is 0xfffe-two-bytes.
```

Comment:
Why is 0xfffe being brought into this now...? This sentence confuses me, I think the grammar is off